pkg/server/registration.go

```diff
@@ -0,0 +1,129 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/server/catalog"
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+	"github.com/spiffe/spire/proto/server/datastore"
+	"golang.org/x/net/context"
+)
+
+//Service is used to register SPIFFE IDs, and the attestation logic that should
+//be performed on a workload before those IDs can be issued.
+type registrationServer struct {
+	l       logrus.FieldLogger
+	catalog catalog.Catalog
+}
+
+//Creates an entry in the Registration table,
+//used to assign SPIFFE IDs to nodes and workloads.
+func (s *registrationServer) CreateEntry(
+	ctx context.Context, request *common.RegistrationEntry) (
+	response *registration.RegistrationEntryID, err error) {
+
+	dataStore := s.catalog.DataStores()[0]
+	createResponse, err := dataStore.CreateRegistrationEntry(
+		&datastore.CreateRegistrationEntryRequest{RegisteredEntry: request},
+	)
+
+	if err != nil {
+		s.l.Error(err)
+		return response, errors.New("Error trying to create entry")
+	}
+
+	return &registration.RegistrationEntryID{Id: createResponse.RegisteredEntryId}, nil
+}
+
+// Implement the business logic of DeleteEntry
+func (s *registrationServer) DeleteEntry(
+	ctx context.Context, request *registration.RegistrationEntryID) (
+	response *common.RegistrationEntry, err error) {
+	return response, err
+}
+
+//Retrieves a specific registered entry
+func (s *registrationServer) FetchEntry(
+	ctx context.Context, request *registration.RegistrationEntryID) (
+	response *common.RegistrationEntry, err error) {
+
+	dataStore := s.catalog.DataStores()[0]
+	fetchResponse, err := dataStore.FetchRegistrationEntry(
+		&datastore.FetchRegistrationEntryRequest{RegisteredEntryId: request.Id},
+	)
+	if err != nil {
+		s.l.Error(err)
+		return response, errors.New("Error trying to fetch entry")
+	}
+	return fetchResponse.RegisteredEntry, nil
+}
+
+// Implement the business logic of UpdateEntry
+func (s *registrationServer) UpdateEntry(
+	ctx context.Context, request *registration.UpdateEntryRequest) (
+	response *common.RegistrationEntry, err error) {
+	return response, err
+}
+
+// Implement the business logic of ListByParentID
```

Comment:
we can remove this comment and probably have a TODO

Reply:
Sure thing. Just TODO?

Reply:
Yes just to make it more obvious that it needs to be implemented